### PR TITLE
Ignore the exception values generated by the winit resize event

### PR DESCRIPTION
**Pull request by jinleili**
**Checklist**

- [x] Run `cargo clippy`.
- [x] Run `cargo clippy --target wasm32-unknown-unknown` if applicable.
- [x] Add change to CHANGELOG.md. See simple instructions inside file.

**Connections**
https://github.com/gfx-rs/wgpu/issues/3915

**Testing**
Tested on macOS 14 Beta 3 and macOS 13.

### CODE REVIEW

File: examples/common/src/framework.rs [hunk 1]
```diff
@@ -307,11 +307,20 @@ fn start<E: Example>(
                     },
                 ..
             } => {
-                log::info!("Resizing to {:?}", size);
-                config.width = size.width.max(1);
-                config.height = size.height.max(1);
-                example.resize(&config, &device, &queue);
-                surface.configure(&device, &config);
+                let max_dimension = adapter.limits().max_texture_dimension_2d;
+                if size.width > max_dimension || size.height > max_dimension {
```
Comment (cwfitzgerald) [inline examples/common/src/framework.rs:313]: I don't get how this is happening in the first place. The limits for max texture dimension should be raised to the maximum supported by the adapter for this very reason.

Comment (jinleili) [reply examples/common/src/framework.rs:313]: This seems to be a bug in winit itself. I have also encountered the same crash when running examples of bevy on my system.

Comment (cwfitzgerald) [reply examples/common/src/framework.rs:313]: What is the dimension given by winit?

Comment (jinleili) [reply examples/common/src/framework.rs:313]: u32::MAX, only happened on macOS Sonoma.

Comment (cwfitzgerald) [reply examples/common/src/framework.rs:313]: The fuck? Can you file a bug with winit, then note that that bug is the reason for this in a comment in the code.

Comment (jinleili) [reply examples/common/src/framework.rs:313]: Sure.

Comment (jinleili) [reply examples/common/src/framework.rs:313]: Someone already filed this bug two weeks ago: https://github.com/rust-windowing/winit/issues/2876

Comment (cwfitzgerald) [reply examples/common/src/framework.rs:313]: Alright, if you can just add the reasoning to the code and then this should be good.

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**